src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -36,30 +36,30 @@ export class AwsSamDebugConfigurationProvider implements vscode.DebugConfigurati
         debugConfiguration: AwsSamDebuggerConfiguration,
         token?: vscode.CancellationToken
     ): Promise<AwsSamDebuggerConfiguration | undefined> {
-        let invalidityPair: [boolean, string | undefined] = generalDebugConfigValidation(debugConfiguration)
+        let validityPair: { isValid: boolean; message?: string } = generalDebugConfigValidation(debugConfiguration)
 
-        if (invalidityPair[0]) {
-            if (invalidityPair[1]) {
-                vscode.window.showErrorMessage(invalidityPair[1])
+        if (!validityPair.isValid) {
```

Comment:
This logic will swallow any warning message due to reassignment down below.

Reply:
I'm not sure I understand what you're saying?...if this is hit, we're outputting a message if it's there, returning `undefined` and not proceeding.

As of now, this isn't handling any warning-but-not-error messages. We can tweak that if the time comes.

Reply:
This was a misunderstanding on my part -- templateDebugConfigValidation is the method that returns errors and warnings